### ISSUE TITLE
Kia Niro 2019 - "Check FCA" warning when engaging openpilot

### ISSUE THREAD

**Issue by DerGue69** [commaai/openpilot#25221]
### Describe the bug

I have a Kia Niro EV 2019 (EU - Austria - 1st Edition). A "Hyundai H" Harness is my connector.
After installin stock OP my car was not recognized. I had to add values to values.py from my cars fingerprint (see additional info)
(I thought My Niro EV is already compatible with OpenPilot)

After that OP started working with my car. But everytime I activate OP I get a "Check FCA ( Forward Coll. Avoidance Assist)" warning. 

I recorded a route and made public.
https://cabana.comma.ai/?route=6111bd7df5043394%7C2022-07-20--14-47-58&exp=1689861829&sig=9eylZ8Csg1lvrMr6yneTzSZbSC7qN5ZjymwZ2jOTkVM%3D&max=-1&url=https%3A%2F%2Fchffrprivate.azureedge.net%2Fchffrprivate3%2Fv2%2F6111bd7df5043394%2Fc7e80e23a0c545d687808655907ee3ee_2022-07-20--14-47-58

Please help me to get rid of this error ;)

![image](https://user-images.githubusercontent.com/96002277/180004334-6e2aa636-f70d-49ec-90a5-5fc3a8c6dd47.png)

### Which car does this affect?

Kia Niro EV 2019 (EU - Austria - 1st Edition)

### Provide a route where the issue occurs

6111bd7df5043394%7C2022-07-20--14-47-58

### openpilot version

0.8.14

### Additional info

![image](https://user-images.githubusercontent.com/96002277/180004251-9bf3ca9c-be34-4e08-9088-809c5e9646a8.png)

'carFingerprint': 'KIA NIRO EV 2020',
               'carFw': [{'address': 2004,
                          'ecu': 'eps',
                          'fwVersion': b'\xf1\x00DE  MDPS C 1.00 1.04 56310Q4100\x00 4DEEC104',
                          'responseAddress': 0,
                          'subAddress': 0},
                         {'address': 2000,
                          'ecu': 'fwdRadar',
                          'fwVersion': b'\xf1\x00DEev SCC F-CUP      1.00 1.02 96400-Q4000         ',
                          'responseAddress': 0,
                          'subAddress': 0},
                         {'address': 1988,
                          'ecu': 'fwdCamera',
                          'fwVersion': b'\xf1\x00DEE MFC  AT EUR LHD 1.00 1.03 95740-Q4000 180821',
                          'responseAddress': 0,
                          'subAddress': 0}],

**Comment by sshane** [comment]
Hi, can you try the branch on this PR? https://github.com/commaai/openpilot/pull/25223

Enter https://installer.comma.ai/commaai/niro-ev as the set up URL after uninstalling openpilot

**Comment by sshane** [comment]
I see you removed your car from that list, was it causing you problems?

**Comment by DerGue69** [comment]
I tried this both things from Discord Chat - But I had no success. (see below)
I will try  https://installer.comma.ai/commaai/niro-ev tomorrow and give you an update asap.


_sunnyhaibin — Yesterday at 9:31 PM
But remove CAR.KIA_NIRO_EV from this line: https://github.com/commaai/openpilot/blob/master/selfdrive/car/hyundai/hyundaican.py#L19
after that, reboot
Just to test
Steering is fine
Image
also worth removing from this: https://github.com/commaai/openpilot/blob/master/selfdrive/car/hyundai/carcontroller.py#L132_

**Comment by sshane** [comment]
Btw do you also trying this branch after? https://github.com/commaai/openpilot/pull/25122

It's a fix for the VIN queries not returning correctly, and it happens your car is affected

**Comment by DerGue69** [comment]
> 

Installed "niro-ev" branch:

- Car was recognized automatically. No need to add fingerprint.
- When engaging OP, again I get the message "FCA ....".
- I shared the route  >> 6111bd7df5043394%7C2022-07-21--09-36-37 << for troubleshooting.
![image](https://user-images.githubusercontent.com/96002277/180159533-9258703e-c628-474b-ab57-7412264d4924.png)


https://cabana.comma.ai/?route=6111bd7df5043394%7C2022-07-21--09-36-37&exp=1689925644&sig=K2wVYIUymbPoaDHnAEK6AwvNuUcYMOIVJ6ihLzTSIHo%3D&max=1&url=https%3A%2F%2Fchffrprivate.azureedge.net%2Fchffrprivate3%2Fv2%2F6111bd7df5043394%2F05fad89595576b1e986e75ebb6623f03_2022-07-21--09-36-37

**Comment by DerGue69** [comment]
I updated the vin.py like in https://github.com/commaai/openpilot/pull/25122 as requested.
I have the same issue. FCA warning.
Shared affected route at: 6111bd7df5043394%7C2022-07-21--10-27-05


https://cabana.comma.ai/?route=6111bd7df5043394%7C2022-07-21--10-27-05&exp=1689928255&sig=LsG%2FeJVfRaHna2668JbOSzOqGHUU4kWGVNt49%2FWbLNc%3D&max=-1&url=https%3A%2F%2Fchffrprivate.azureedge.net%2Fchffrprivate3%2Fv2%2F6111bd7df5043394%2F6f020ce2d536d1c8ccde85ebfd2a5b95_2022-07-21--10-27-05

**Comment by sshane** [comment]
Can you try the branch on this PR? https://github.com/commaai/openpilot/pull/25237 Please make sure to actually check out the branch cleanly, not modify another branch

**Comment by adeebshihadeh** [comment]
@DerGue69 have you had a chance to try the branch?

**Comment by sshane** [comment]
Yes, we've been chatting on Discord. It works to hide the alerts, but his car is still sending abnormal signals around related to FCA and FCW. However he's still able to get FCW so I'm going to confirm with Greg how we should merge this.

**Comment by DerGue69** [comment]
> @DerGue69 have you had a chance to try the branch?

Yes. Shane and I did lots of tests. This branch gets rid of the annoying warning FCA warning.
When driving too fast to a standing car, I still get a collision warning in the car.

**Comment by sshane** [comment]
@DerGue69 have you tried removing/adding your car to the various lists in the same function we modify the fusion state?

**Comment by DerGue69** [comment]
@sshane: What exactly do you mean with "removing/adding your car to the various lists"?

In openpilot/selfdrive/car/hyundai/hyundaican.py ...
in the section of if car_fingerprint in Kia Niro ... 
I added:

    # Some cars show an FCA fault when forwarding CF_Lkas_FusionState. Hide FCA Fault:
    values["CF_Lkas_FusionState"] = 0

![image](https://github.com/commaai/openpilot/assets/96002277/0913482c-b8dd-4579-9236-5a9ebeda6365)

I still have no idea, what the "CF_Lkas_FusionState" is - but it hides the FCA warning completely.
I cannot perform a real-world crash test - to check if the actual FCA is working. (lack of inflateable car)

**Comment by Mirukuuu** [comment]
Hi, I just got my comma 3X and connected it to a Kia Niro EV 2019 in Europe and have the same warning as described. Has there been any further progress on this issue?

**Comment by DerGue69** [comment]
Hey Mirukuuu, greetings from Vienna.
Shane and I tried lots of things. The last thing I did, I watched Cabana Code like in Matrix to find some clues.
I even made a crash test with my brother to get some data...

Long story short:
I made my own branch as a 99.99% clone from Master.
The 0,01 % difference is one line of code in : openpilot/tree/master/selfdrive/car/hyundai/hyundaican.py
**### values["CF_Lkas_FusionState"] = 0**

This makes the error vanish. The only thing "we" don't know, if the real FCA (FrontCollissionAssistent) is still working with that code added. (That's why I was planning a crash-test)

I use my 99.99% clone since 60k+ km without any issues and warnings:
https://github.com/DerGue69/openpilot/tree/master

**Comment by Mirukuuu** [comment]
Thanks a lot DerGue69 for the quick reply and the extensive explanation! I'll see if I can try this out soon, seems easy enough.

**Comment by DerGue69** [comment]
If you need help/tips ... let me know.

**Comment by DerGue69** [comment]
> Thanks a lot DerGue69 for the quick reply and the extensive explanation! I'll see if I can try this out soon, seems easy enough.

Any updates on your integration?

**Comment by DerGue69** [comment]
> Thanks a lot DerGue69 for the quick reply and the extensive explanation! I'll see if I can try this out soon, seems easy enough.

Any updates?
